<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MME</class>
 <widget class="QWidget" name="MME">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>480</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>20</width>
    <height>20</height>
   </size>
  </property>
  <property name="baseSize">
   <size>
    <width>800</width>
    <height>480</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_5">
   <item>
    <widget class="QStackedWidget" name="stackedWidget">
     <property name="currentIndex">
      <number>1</number>
     </property>
     <widget class="QWidget" name="mainPage">
      <layout class="QVBoxLayout" name="verticalLayout_8">
       <item>
        <widget class="QSplitter" name="buttonsSplitter">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="handleWidth">
          <number>25</number>
         </property>
         <widget class="QWidget" name="layoutWidget">
          <layout class="QHBoxLayout" name="horizontalLayout_3">
           <item>
            <widget class="QSplitter" name="logSplitter">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="handleWidth">
              <number>15</number>
             </property>
             <widget class="QWidget" name="layoutWidget">
              <layout class="QVBoxLayout" name="verticalLayout_2">
               <item>
                <layout class="QVBoxLayout" name="matrixLayout"/>
               </item>
               <item>
                <layout class="QHBoxLayout" name="horizontalLayout_5">
                 <item>
                  <widget class="QLineEdit" name="command">
                   <property name="styleSheet">
                    <string notr="true"/>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QPushButton" name="pushButton">
                   <property name="maximumSize">
                    <size>
                     <width>27</width>
                     <height>16777215</height>
                    </size>
                   </property>
                   <property name="text">
                    <string>C</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
              </layout>
             </widget>
             <widget class="QTextBrowser" name="log">
              <property name="styleSheet">
               <string notr="true"/>
              </property>
             </widget>
            </widget>
           </item>
           <item>
            <layout class="QVBoxLayout" name="verticalLayout_4">
             <item>
              <layout class="QVBoxLayout" name="rowLayout">
               <property name="spacing">
                <number>2</number>
               </property>
               <property name="sizeConstraint">
                <enum>QLayout::SetMaximumSize</enum>
               </property>
              </layout>
             </item>
             <item>
              <spacer name="verticalSpacer">
               <property name="orientation">
                <enum>Qt::Vertical</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>20</width>
                 <height>40</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="layoutWidget">
          <layout class="QVBoxLayout" name="verticalLayout">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout">
             <item>
              <widget class="QLineEdit" name="rowEdit">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="maximumSize">
                <size>
                 <width>35</width>
                 <height>35</height>
                </size>
               </property>
               <property name="baseSize">
                <size>
                 <width>0</width>
                 <height>0</height>
                </size>
               </property>
               <property name="styleSheet">
                <string notr="true"/>
               </property>
               <property name="text">
                <string>3</string>
               </property>
               <property name="alignment">
                <set>Qt::AlignCenter</set>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="label">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="text">
                <string>Ã—</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="colEdit">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="maximumSize">
                <size>
                 <width>35</width>
                 <height>35</height>
                </size>
               </property>
               <property name="styleSheet">
                <string notr="true"/>
               </property>
               <property name="text">
                <string>3</string>
               </property>
               <property name="alignment">
                <set>Qt::AlignCenter</set>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="clearButton">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="font">
                <font>
                 <pointsize>16</pointsize>
                </font>
               </property>
               <property name="text">
                <string>Clear</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="undoButton">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="font">
                <font>
                 <pointsize>16</pointsize>
                </font>
               </property>
               <property name="text">
                <string>Undo</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QGridLayout" name="gridLayout">
             <property name="topMargin">
              <number>9</number>
             </property>
             <property name="bottomMargin">
              <number>9</number>
             </property>
             <property name="verticalSpacing">
              <number>12</number>
             </property>
             <item row="0" column="0">
              <widget class="QPushButton" name="plusButton">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="font">
                <font>
                 <pointsize>16</pointsize>
                </font>
               </property>
               <property name="text">
                <string>+</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QPushButton" name="mulButton">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="font">
                <font>
                 <pointsize>16</pointsize>
                </font>
               </property>
               <property name="text">
                <string>*</string>
               </property>
              </widget>
             </item>
             <item row="0" column="2">
              <widget class="QPushButton" name="identityButton">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="font">
                <font>
                 <pointsize>16</pointsize>
                </font>
               </property>
               <property name="text">
                <string>+E</string>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QPushButton" name="minusButton">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="font">
                <font>
                 <pointsize>16</pointsize>
                </font>
               </property>
               <property name="text">
                <string>-</string>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QPushButton" name="divButton">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="font">
                <font>
                 <pointsize>16</pointsize>
                </font>
               </property>
               <property name="text">
                <string>/</string>
               </property>
              </widget>
             </item>
             <item row="1" column="2">
              <widget class="QPushButton" name="detButton">
               <property name="font">
                <font>
                 <pointsize>16</pointsize>
                </font>
               </property>
               <property name="text">
                <string>Det</string>
               </property>
              </widget>
             </item>
             <item row="2" column="0">
              <widget class="QPushButton" name="oneButton">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="font">
                <font>
                 <pointsize>16</pointsize>
                </font>
               </property>
               <property name="text">
                <string>1</string>
               </property>
              </widget>
             </item>
             <item row="2" column="1">
              <widget class="QPushButton" name="threeButton">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="font">
                <font>
                 <pointsize>16</pointsize>
                </font>
               </property>
               <property name="text">
                <string>3</string>
               </property>
              </widget>
             </item>
             <item row="2" column="2">
              <widget class="QPushButton" name="bareissButton">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="sizeIncrement">
                <size>
                 <width>0</width>
                 <height>0</height>
                </size>
               </property>
               <property name="font">
                <font>
                 <pointsize>16</pointsize>
                </font>
               </property>
               <property name="text">
                <string>Bareiss</string>
               </property>
              </widget>
             </item>
             <item row="3" column="0">
              <widget class="QPushButton" name="twoButton">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="font">
                <font>
                 <pointsize>16</pointsize>
                </font>
               </property>
               <property name="text">
                <string>2</string>
               </property>
              </widget>
             </item>
             <item row="3" column="1">
              <widget class="QPushButton" name="fourButton">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="font">
                <font>
                 <pointsize>16</pointsize>
                </font>
               </property>
               <property name="text">
                <string>4</string>
               </property>
              </widget>
             </item>
             <item row="3" column="2">
              <widget class="QPushButton" name="transposeButton">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="font">
                <font>
                 <pointsize>16</pointsize>
                </font>
               </property>
               <property name="text">
                <string>T</string>
               </property>
              </widget>
             </item>
             <item row="4" column="2">
              <widget class="QPushButton" name="mulLastColButton">
               <property name="font">
                <font>
                 <pointsize>16</pointsize>
                </font>
               </property>
               <property name="text">
                <string>*lastCol</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QPushButton" name="swapButton">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="font">
              <font>
               <pointsize>16</pointsize>
              </font>
             </property>
             <property name="text">
              <string>Swap</string>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_7">
             <item>
              <widget class="QPushButton" name="goButton">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="font">
                <font>
                 <pointsize>16</pointsize>
                </font>
               </property>
               <property name="text">
                <string>Go</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="nextLayerButton">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="font">
                <font>
                 <pointsize>16</pointsize>
                </font>
               </property>
               <property name="text">
                <string>-&gt;</string>
               </property>
               <property name="flat">
                <bool>false</bool>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="otherPage">
      <layout class="QVBoxLayout" name="verticalLayout_7">
       <item>
        <widget class="QSplitter" name="helpSplitter">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="handleWidth">
          <number>25</number>
         </property>
         <widget class="QGroupBox" name="textMatrixGroupBox">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="title">
           <string>Matrix from text</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_3">
           <item>
            <widget class="QPlainTextEdit" name="textMatrixEdit">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_6">
             <item>
              <widget class="QPushButton" name="clearTexMatrixButton">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="font">
                <font>
                 <pointsize>20</pointsize>
                </font>
               </property>
               <property name="text">
                <string>Clear</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="electricButton">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="font">
                <font>
                 <pointsize>20</pointsize>
                </font>
               </property>
               <property name="text">
                <string>Electric</string>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="horizontalSpacer">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeType">
                <enum>QSizePolicy::Expanding</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
             <item>
              <widget class="QPushButton" name="submitTextMatrixButton">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="font">
                <font>
                 <pointsize>20</pointsize>
                </font>
               </property>
               <property name="text">
                <string>Submit</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
         <widget class="QScrollArea" name="scrollArea">
          <property name="widgetResizable">
           <bool>true</bool>
          </property>
          <widget class="QWidget" name="scrollAreaWidgetContents">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
             <width>79</width>
             <height>442</height>
            </rect>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_6">
            <item>
             <widget class="QLabel" name="helpLabel">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string/>
              </property>
              <property name="textFormat">
               <enum>Qt::RichText</enum>
              </property>
              <property name="alignment">
               <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
              </property>
              <property name="wordWrap">
               <bool>true</bool>
              </property>
              <property name="openExternalLinks">
               <bool>true</bool>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>command</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>283</x>
     <y>173</y>
    </hint>
    <hint type="destinationlabel">
     <x>245</x>
     <y>171</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>clearTexMatrixButton</sender>
   <signal>clicked()</signal>
   <receiver>textMatrixEdit</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>72</x>
     <y>441</y>
    </hint>
    <hint type="destinationlabel">
     <x>72</x>
     <y>396</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
